include/boost/gil/io/device.hpp

```diff
@@ -435,7 +419,7 @@ class istream_device
 
     void write(const byte_t*, std::size_t)
     {
-        io_error( "Bad io error." );
+        io_error( "istream_device: Bad io error." );
```

Comment:
I appreciate you've taken care of such details as clarifying the exception messages here and in other places.

Reply:
The JPEG file that was causing the assert was a zero-length file with extension .jpg , easy to be created by "touch zerolengthfile.jpg".

Reply:
Yes, easy. I wasn't certain if it is zero-file or perhaps a degenerate file with JPEG signature only or something alike.